### PR TITLE
Add scheduled changefeed docs to SQL and CDC

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-6790

This adds products docs for the scheduled changefeed feature, including:

- New SQL page for `CREATE SCHEDULE FOR CHANGEFEED`
- Updates to the `SHOW`, `PAUSE`, and `RESUME` schedule pages.
- Updates and new example to the Export Data with Changefeeds page

Not included in this PR is any updates to the privileges section around 23.1 work on fine-grained permissions. This is to come in a follow-up PR for the permission roadmap feature. 

## Preview

https://deploy-preview-16257--cockroachdb-docs.netlify.app/docs/dev/create-schedule-for-changefeed.html

### CODE REVIEW

File: v23.1/create-schedule-for-changefeed.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+docs_area: reference.sql
+---
+
+{% include_cached new-in.html version="v23.1" %} `CREATE SCHEDULE FOR CHANGEFEED` allows you to create a scheduled changefeed to export data out of CockroachDB. Scheduled changefeeds have the scale, observability, and endpoint sink options that changefeed jobs include, with the convenience of setting a regular schedule. A changefeed job created with `CREATE SCHEDULE FOR CHANGEFEED` performs a one-time table scan using the [initial scan](create-changefeed.html#initial-scan) functionality to create an export of your table data.
+
+For more detail on using changefeeds to create an export of your table data, see [Export Data with Changefeeds](export-data-with-changefeeds.html).
+
+## Required privileges
+
+{% include {{ page.version.version }}/cdc/privilege-model.md %}
```
Comment (kathancox) [inline v23.1/create-schedule-for-changefeed.md:14]: These privilege sections are pulled in from the `CREATE CHANGEFEED` page, which was updated with the permission work in v22.2. There may be some changes needed to this that I will complete with the docs feature work for 23.1 fine-grained permissions in Feb/Mar while the 23.1 docs are still in `dev`.

**Comment by kathancox** [comment]
cc @amruss

File: v23.1/create-schedule-for-changefeed.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+</div>
+
+### Parameters
+
+Parameter | Description
+----------+------------
+`IF NOT EXISTS` | A scheduled changefeed should not be created if the `schedule_label` already exists. You will receive an error if the schedule label already exists, or if `schedule_label` is not defined when using `IF NOT EXISTS`.
+`schedule_label` | The name for the scheduled changefeed. This is optional and does not need to be unique. If you do not define a name, the label will default to `CHANGEFEED` with the timestamp of when you created the schedule. 
+`changefeed_targets` | The tables to target with the changefeed. For example, `movr.users, movr.rides`.
+`changefeed_sink` | A URI to authenticate to the [changefeed sink](changefeed-sinks.html).
```
Comment (jayshrivastava) [inline v23.1/create-schedule-for-changefeed.md:33]: The description can just be "changefeed sink URI", with the hyperlink.

File: v23.1/drop-schedules.md [hunk 3]
```diff
@@ -65,6 +63,7 @@ In this example, all schedules with the label `schedule_database` are dropped.
 - [Manage a Backup Schedule](manage-a-backup-schedule.html)
 - [`BACKUP`](backup.html)
 - [`RESTORE`](restore.html)
+- [`CREATE CHANGEFEED`](create-changefeed.html)
```
Comment (jayshrivastava) [inline v23.1/drop-schedules.md:66]: Maybe link the "Create Schedule for Changefeed" table.

**Review by jayshrivastava — APPROVED** [review]
Just a few suggestions. Otherwise this LGTM!

File: v23.1/change-data-capture-overview.md [hunk 4]
```diff
@@ -19,8 +19,8 @@ The main feature of CDC is the changefeed, which targets an allowlist of tables,
...
-| Watches one or multiple tables in a comma-separated list. Emits every change to a "watched" row as a record. | Watches one or multiple tables in a comma-separated list. Emits every change to a "watched" row as a record in a <br> configurable format (`JSON` or Avro) to a [configurable sink](changefeed-sinks.html) (e.g., [Kafka](https://kafka.apache.org/)). |
+| Create with [`EXPERIMENTAL CHANGEFEED FOR`](changefeed-for.html). | Create with [`CREATE CHANGEFEED`](create-changefeed.html).<br>Use `CREATE CHANGEFEED` with [CDC transformations](cdc-transformations.html) to define the emitted change data.<br>Create a scheduled changefeed with [`CREATE SCHEDULE FOR CHANGEFEED`](create-schedule-for-changefeed.html). |
+| Watches one or multiple tables in a comma-separated list. Emits every change to a "watched" row as a record. | Watches one or multiple tables in a comma-separated list. Emits every change to a "watched" row as a record in a <br> configurable format (JSON, CSV, Avro) to a [configurable sink](changefeed-sinks.html) (e.g., [Kafka](https://kafka.apache.org/)). |
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/change-data-capture-overview.md]: Done. Thanks for noticing this!

File: v23.1/create-schedule-for-changefeed.md (hunk 2 above)
Comment (kathancox) [reply v23.1/create-schedule-for-changefeed.md]: Edited!

File: v23.1/change-data-capture-overview.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/change-data-capture-overview.md]: I haven't added Parquet to this yet as @amruss mentioned the performance work around this format. I will double check whether we should document this yet. Thanks for bringing up though!

File: v23.1/drop-schedules.md (hunk 3 above)
Comment (kathancox) [reply v23.1/drop-schedules.md:66]: Done.

File: v23.1/export-data-with-changefeeds.md [hunk 5]
```diff
@@ -37,6 +36,19 @@ SHOW CHANGEFEED JOB {job ID};
...
+~~~
+
+The [schedule options](create-schedule-for-changefeed.html#schedule-options) control the schedule's behavior:
+
+- If it runs into a failure, `on_execution_failure=retry` will ensure that the schedule retries the changefeed immediately. 
+- If the previous scheduled changefeed is still running, `on_previous_running=start` will start a new changefeed at the defined cadence.
+
 ## See also
 
 - [Changefeed Messages](changefeed-messages.html)
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/export-data-with-changefeeds.md:54]: Done.

**Review by amruss — APPROVED** [review]
lgtm

File: v23.1/create-schedule-for-changefeed.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+You can include the changefeed options listed on the [`CREATE CHANGEFEED`](create-changefeed.html#options) page to modify the behavior of your changefeed. The following options are **not** compatible with scheduled changefeeds:
+
+- `diff`
+- `end_time`
+- `mvcc_timestamp`
+- `resolved`
+- `updated`
+
+Scheduled changefeeds have the [`initial_scan = 'only'`](create-changefeed.html#initial-scan) or `initial_scan_only` option included implicitly. You cannot specify `initial_scan` as `'yes'` or `'no'`.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/create-schedule-for-changefeed.md:51]: Ah yeah, good point. We v briefly mention the old syntax on the `CREATE CHANGEFEED` page, but that is only to say it is still supported right now. I will remove `initial_scan_only` from here.

File: v23.1/create-schedule-for-changefeed.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+### Parameters
+
+Parameter | Description
+----------+------------
+`IF NOT EXISTS` | A scheduled changefeed should not be created if the `schedule_label` already exists. You will receive an error if the schedule label already exists, or if `schedule_label` is not defined when using `IF NOT EXISTS`.
+`schedule_label` | The name for the scheduled changefeed. This is optional and does not need to be unique. If you do not define a name, the label will default to `CHANGEFEED` with the timestamp of when you created the schedule. 
+`changefeed_targets` | The tables to target with the changefeed. For example, `movr.users, movr.rides`.
+`changefeed_sink` | The [changefeed sink URI](changefeed-sinks.html).
+`changefeed_option` | The [options](create-changefeed.html#options) to control the behavior of your changefeed. For example, `WITH format = csv, full_table_name`. See [Changefeed options](#changefeed-options) for a list of exceptions.
```
Comment (andf-crl) [inline v23.1/create-schedule-for-changefeed.md:34]: nit: `exceptions` --> `available options`

File: v23.1/create-schedule-for-changefeed.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+This scheduled changefeed filters for the usage of promotion codes in the [`movr`](movr.html) database and sends the changefeed messages on a daily basis:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE SCHEDULE FOR CHANGEFEED INTO 'external://kafka-sink' AS SELECT user_id, usage_count FROM movr.user_promo_codes WHERE usage_count > 1 RECURRING '@daily' WITH SCHEDULE OPTIONS first_run=now, on_execution_failure=reschedule, on_previous_running=skip;
+~~~
+
+### View scheduled changefeed details
+
+To [show](show-schedules.html) all scheduled changefeeds:
```
Comment (andf-crl) [inline v23.1/create-schedule-for-changefeed.md:93]: I live the linked verbs for examples that introduce a new verb statement. Works well!

File: v23.1/create-schedule-for-changefeed.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+### Create a scheduled changefeed
+
+The following statement sets up a scheduled changefeed named `users_rides_nightly` that will send changefeed messages in `CSV` format 1 minute past midnight every night. As soon as the statement is run, the first changefeed run will execute immediately:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE SCHEDULE users_rides_nightly FOR CHANGEFEED INTO 'external://kafka-sink' WITH format=csv RECURRING '1 0 * * *' WITH SCHEDULE OPTIONS first_run=now, on_execution_failure=retry, on_previous_running=skip;
+~~~
+
+If the scheduled changefeed runs into an error, it will retry the changefeed immediately. The changefeed will skip a new changefeed run if the previous run is still running at the next scheduled time.
```
Comment (andf-crl) [inline v23.1/create-schedule-for-changefeed.md:76]: nit: I really like the bulleted breakdown of what this command does that you've used on `v23.1/export-data-with-changefeeds.md` lines 47-50. Could that be replicated here for clarity? Notably, the pairing of, say, "`on_previous_running=skip` means X will happen" helps to parse the whole command at a glance. Optional!

**Review by andf-crl — APPROVED** [review]
LGTM, just some small optional thoughts. Thanks!

File: v23.1/create-schedule-for-changefeed.md (hunk 8 above)
Comment (kathancox) [reply v23.1/create-schedule-for-changefeed.md:96]: Oh good!

File: v23.1/create-schedule-for-changefeed.md (hunk 7 above)
Comment (kathancox) [reply v23.1/create-schedule-for-changefeed.md]: Done!

File: v23.1/create-schedule-for-changefeed.md (hunk 9 above)
Comment (kathancox) [reply v23.1/create-schedule-for-changefeed.md]: Done, good call!

File: v23.1/change-data-capture-overview.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kathancox) [reply v23.1/change-data-capture-overview.md]: To wrap up this conversation, I discussed this with Abbey and she confirmed to wait on Parquet for now.

**Pull request merged**